src/utils/server-utils.ts

```diff
@@ -225,6 +231,16 @@ async function getPointDataCoverage(layer: PointDataLayerProps) {
   return possibleDates;
 }
 
+async function getNSOAvailableDates(layer: NSOLayerProps) {
+  const url = layer.dateUrl!;
```

Comment:
why can we assume this is not null? IMO might be safer to try stick to the TS definitions (add a null check)

Reply:
Also, is it possible to link this with the `getPossibleDatesForLayer` which we already have?